server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/server.js

```diff
@@ -58,115 +57,80 @@ function refreshDetailTable() {
 }
 
 /**
- * Generates the server history table
+ * Populates the All Time Tablet Operations table
  */
 function refreshHistoryTable() {
-
-  clearTableBody('opHistoryDetails');
-
   var data = sessionStorage.server === undefined ?
       [] : JSON.parse(sessionStorage.server);
 
   if (data.length === 0 || data.allTimeTabletResults === undefined) {
-    var row = [];
-
-    row.push(createEmptyRow(8, 'Empty'));
-
-    $('<tr/>', {
-      html: row.join('')
-    }).appendTo('#opHistoryDetails tbody');
+    clearAllTableCells("opHistoryDetails");
   } else {
     var totalTimeSpent = 0;
     $.each(data.allTimeTabletResults, function(key, val) {
       totalTimeSpent += val.timeSpent;
     });
 
     $.each(data.allTimeTabletResults, function(key, val) {
-      var row = [];
-
-      row.push(createFirstCell(val.operation, val.operation));
-
-      row.push(createRightCell(val.success, bigNumberForQuantity(val.success)));
-
-      row.push(createRightCell(val.failure,
-          bigNumberForQuantity(val.failure)));
-
-      row.push(createRightCell((val.avgQueueTime == null ?
-          '-' : val.avgQueueTime * 1000.0),
-          (val.avgQueueTime == null ?
-          '&mdash;' : timeDuration(val.avgQueueTime * 1000.0))));
-
-      row.push(createRightCell((val.queueStdDev == null ?
-          '-' : val.queueStdDev * 1000.0),
-          (val.queueStdDev == null ?
-          '&mdash;' : timeDuration(val.queueStdDev * 1000.0))));
-
-      row.push(createRightCell((val.avgTime == null ?
-          '-' : val.avgTime * 1000.0),
-          (val.avgTime == null ?
-          '&mdash;' : timeDuration(val.avgTime * 1000.0))));
-
-      row.push(createRightCell((val.stdDev == null ?
-          '-' : val.stdDev * 1000.0),
-          (val.stdDev == null ?
-          '&mdash;' : timeDuration(val.stdDev * 1000.0))));
-
-      row.push(createRightCell(((val.timeSpent / totalTimeSpent) * 100),
-          '<div class="progress"><div class="progress-bar"' +
-          ' role="progressbar" style="min-width: 2em; width:' +
-          Math.floor((val.timeSpent / totalTimeSpent) * 100) +
-          '%;">' + Math.floor((val.timeSpent / totalTimeSpent) * 100) +
-          '%</div></div>'));
-
-      $('<tr/>', {
-        html: row.join('')
-      }).appendTo('#opHistoryDetails tbody');
-
+      // use the first 5 characters of the operation for the jquery selector
+      var rowId = "#" + val.operation.slice(0, 5) + "Row";
+      console.log("Populating rowId " + rowId + " with " + val.operation + " data");
+
+      $(rowId).append("<td>" + val.operation + "</td>");
+      $(rowId).append("<td>" + bigNumberForQuantity(val.success) + "</td>");
+      $(rowId).append("<td>" + bigNumberForQuantity(val.failure) + "</td>");
+
+      if (val.avgQueueTime == null) {
+        $(rowId).append(dashCell);
+      } else {
+        $(rowId).append("<td>" + timeDuration(val.avgQueueTime * 1000.0) + "</td>");
+      }
```

Comment:
This logic is repeated a lot. Could put this in a separate function:

```js
function appendDurationToCell(rowId, value) {
  let v = dashCell;
  if (value != null) {
    v = "<td>" + timeDuration(val.avgQueueTime * 1000.0) + "</td>";
  }
  $(rowId).append(v);
}
// ...
appendDurationToCell(rowId, val.avgQueueTime);
appendDurationToCell(rowId, val.queueStdDev);
```

Reply:
Added new method in cf930c4d5f75fd2f01b48d55dcc21edda8e49f49